### DISCUSSION TITLE
Import from Firefox not available

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#881] [Ideas]
### What feature would you like to see?

I supposed import from Firefox would be available, but it's not. Why?

### What platform are you seeing the problem on?

Linux

### Additional context

Linux AppImage.
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/221_

**Comment by RafaelLinux** [comment]
Totally agree. There is no need to have

I don't have a Firefox account to synchronise, because I don't need it (I don't use the same bookmarks at work as I do at home or on my smartphone) and I don't want it either, so I still don't understand how it's possible that I can't import from Firefox but I can from other browsers.

**Comment by Arctomachine** [comment]
Normal browser import carries over existing sessions. It is seamless transition. Once done, user can continue using it without additional steps.
Firefox account only applies to bookmarks and plugins (which it enables by default even if they were disabled in old browser). User have to manually log into each site (even with password manager it is long and tedious process) before using browser normally. So while sync is good feature on its own, it is not replacement for import.

**Comment by ZombieLurker** [comment]
Not working for me either. Tried with Flatpak and Appimage in Arch Linux.

**Reply by RafaelLinux** [reply]
It's not that it doesn't work for you, my friend, it's that it's not implemented!!!

**Comment by kittybwained** [comment]
yeah, this was pretty odd to me when i installed zen
it wasn't a big deal cause i could just export my passwords to CSV and my bookmarks to HTML and then import those, but it was a little annoying

**Comment by AurelienConte** [comment]
I'm bumping this discussion too since a lot of us are coming from firefox to try zen and are not using firefox sync for privacy reasons.

**Comment by Franz1241** [comment]
I have 3 different profiles and only my personal profile has a Mozilla account.

I would like to just copy and paste the .mozilla into .zen and have it working.

How can we approach this?

**Comment by CroaBeast** [comment]
bumping, need this feature

**Comment by Juts2** [comment]
Also requesting this feature.

**Comment by tobozo** [comment]
Hi Zen team, 

I downloaded the linux/zipped version and canceled the first run at this exact step, I'm subscribing to this discussion and will just wait until this is implemented to give it a try.

Thanks for promoting browser diversity! :heart_eyes: 

![image](https://github.com/user-attachments/assets/58531a75-4312-4da1-a2df-46c723f26ca6)

**Comment by Xerxes004** [comment]
Hey all, I have a workaround that might work for you depending on how much you hate Edge and Chrome.

I got around this issue by first syncing my Firefox profile data with Edge (signed out!), closing Edge, restarting Zen, then importing from Edge into Zen.

No online sync required.

Hope this helps!

CC: #221

**Reply by RafaelLinux** [reply]
Of the last few months, yours is the best post of the many forums I visit, because it is not only very useful, but also makes a very funny use of irony. I think it's a very good temporary solution.

If you don't mind, I'd like to add that to make everyone happy with the temporary solution without feeling bad about using ‘evil’ applications, they could use Chromium instead of the ones you mention. It should work anyway.

THANKS FOR YOUR CONTRIBUTION

**Comment by londek** [comment]
Hey,
I found a workaround without Edge to import (most if not all) data from Firefox to Zen.

Locate your Firefox profile directory
1. Open Firefox
2. Open hamburger menu
3. ->Help
4. ->More Troubleshooting Information
5. In Application Basics table locate "Profile Folder" and open it.

Locate your Zen profile directory
-> Repeat all steps as for Firefox, but in Zen

Actually migrate:
1. Close both Zen and Firefox.
2. In Firefox profile directory select all files (ctrl+a)
3. NOW IMPORTANT PART: locate file "compatibility.ini" and deselect it (ctrl+left click on it).
4. Paste all files to Zen profile. If prompted make sure to overwrite the files. There might be .lock file that can't be overwritten, feel free to skip it.
5. Open Zen, it will behave as when during fresh start.
6. After that everything should be imported correctly ;3

If you are signed out of services, you are probably missing cookies.
Copy cookies.sqlite from Firefox profile into Zen profile.

<img src="https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExb2Vsdm5nYWh6dDR4d2JvdXlwYm5kMTVycmt6NmN4ZzJwOHowcGdhdCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/sUYN0JCziyLZop4hi0/giphy.webp" height="100px">

**Reply by Nattfisk** [reply]
Worked great thank you!

**Reply by weikequ** [reply]
Worked really well, thanks for sharing!

**Reply by Zlendy** [reply]
Worked like a charm, thanks!

**Reply by abe-101** [reply]
just note that if you install it via flathub
the data location in settings isn't correct and its instead located in 
`~/.var/app/app.zen_browser.zen/.zen/mrhrlfpn.Default (release)`

**Reply by Ultrate** [reply]
for this part "Open Firefox
Open hamburger menu
->Help
->More Troubleshooting Information" you can just type "about:support" on  the search bar and it will take you there thanks for your explanation by the way

**Reply by FifthWit** [reply]
tried that, it just weirly ended up reverting my user.js on normal firefox and while the bookmarks and cookies are correct nothing loads

**Reply by kortschak** [reply]
Did this and it worked _almost_ perfectly. The only issue was that treatment of context menu styling where the text is rendered with low contrast (I'm guessing this is because firefox was rendering them with dark mode despite having light mode set :shrug:). Can someone point me to the file that holds the colour styling for context menus?

**Reply by akoppa** [reply]
Yes, but this way Zen is converted back into Firefox.

**Comment by WandersonAlves** [comment]
I just watched some videos about Zen browser and thought "hmm, I could replace firefox with zen!". It's frustrating to see that I can't, It's more frustrating to know that Zen is built on top of Firefox...

**Reply by Juts2** [reply]
Looks like it has firefox sync, so you can sign in through the settings menu now and sync your firefox settings that way.

**Reply by wolfkin** [reply]
works like gangbusters. still would love a manual import but since I do use Firefox sync that'll be perfect for me at least.

**Comment by sharp2448** [comment]
bumping, I'm also missing this feature.

**Comment by Arctomachine** [comment]
Recently I found that firefox uses profiles that can be used for migration between different versions. You can access it by going to about:profiles in url bar.
When launching browser with selected profile, it carries over everything: history, opened tabs, addons with all settings, cookies and sessions, everything.

As expected, directly using firefox profiles in zen is impossible. But that is what could potentially be used to make migration easier. Dev team, what do you think about using it?

**Reply by 12th-devs** [reply]
Logging in to your Firefox/Mozilla account will do everything you need it too

**Reply by Arctomachine** [reply]
If it really did, half of people here would already have used it by now.

**Comment by tiropictor** [comment]
This is one most odd fail I could imagine. The browser, based on Firefox cannot import ANYTHING from Firefox. Yes, I don't have FF account/cloud/sync whatever they call it, nor I ever planned to create that. Browser is kinda software to show webpages, it's not another goddamn cursed account-driven app that want all your data and password.. And I use Mac OS, so, obviously windows-based workarounds (using Edge? Seriously? just to import bookmarks?) doesn't fit me. Most funny is that FF is not even in "Import Browser Data" list. I think I need to wait a little while devs realize that problem actually exists..

**Reply by tiropictor** [reply]
follow-up: this solution worked for me perfectly:
https://github.com/zen-browser/desktop/discussions/881#discussioncomment-11217246
all opened windows, tabs, all extension, all sessions - everything just landed to Zen Browser as I EXPECT IT TO BE DONE. 
**Is it really that hard to proceed same way while initial setup of Zen Browser**?? (question to devs)

**Comment by AppearedOne** [comment]
I also was really confused about this but ended up using [this workaround from the guides section](https://docs.zen-browser.app/guides/manage-profiles). Though it's insane to me how a Firefox based browser can't do this annoying step automatically.

**Comment by chicagobuss** [comment]
just want to add my support for this.  It's really weird you can import from chromium browsers but not firefox .... on a firefox-based browser!  This seems like a critical feature to implement.

**Comment by solarkraft** [comment]
It's so funny and dumb that Zen can import from Chrome but not from Firefox. [Londek's solution](https://github.com/zen-browser/desktop/discussions/881#discussioncomment-11217246) seems like it'd be somewhat easy to automate.

**Comment by chicagobuss** [comment]
Been over six months - what's the zen team thinking about on this issue?  I installed zen on a new computer today and was gobsmacked to see it still can't import from firefox...

**Comment by EmpathyZeroed** [comment]
so cringe to not have this feature

**Comment by gu3st** [comment]
Having import from Firefox would be greatly appreciated especially after the recent CEO stance of them going further into the AI sloplands.

**Comment by srcnalt** [comment]
Discussed over a year ago and still not available. (Not demanding it, just another Firefox refugee here.)

**Comment by Gamigre** [comment]
As the other said, it's really odd to find that a fork of firefox doesn't have a proper import feature. I wanted a smooth transition, or at least to be able to test it out seamlessly and I'm already looking at scripts and various maneuvers on github not even a minute after installing it. It's disappointing, especially when I reckon most zen users would be people already using firefox and tangentially, not the population with the highest amount of users syncing their data online.

**Comment by KarmaMarazu** [comment]
i recently wanted to check out zen and see how it is as i heard some good things about it but i find it really frustrating that it doesnt have a firefox import function and when i tried the https://github.com/zen-browser/desktop/discussions/881#discussioncomment-11217246 work around it just broke the browser for me and didnt let me open new tabs how is this still an issue after almost 2 years???

**Comment by entrophae** [comment]
My workaround was opening Edge, importing from firefox over there and then opening Zen and importing from Edge
